File: app/src/main/java/com/unnebulous/consultapronta/views/SymptomAdapter.java
package com.unnebulous.consultapronta.views;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.unnebulous.consultapronta.R;

import java.util.List;

public class SymptomAdapter
        extends RecyclerView.Adapter<SymptomAdapter.SymptomViewHolder> {
    private final List<Symptom> symptoms;
    public SymptomAdapter(List<Symptom> symptoms) {this.symptoms = symptoms;
    }

    @NonNull
    @Override
    public SymptomViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_symptom, parent, false);
        return new SymptomViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SymptomViewHolder holder, int position) {

        Symptom symptom = symptoms.get(position);

        holder.intensityText.setText(holder.itemView.getContext().getString(R.string.intensity, symptom.getIntensity()));

        holder.descriptionText.setText(symptom.getDescription());

        holder.dateText.setText(symptom.getDate());

        holder.locationText.setText(holder.itemView.getContext().getString(R.string.location, symptom.getLocation()));
    }

    @Override
    public int getItemCount() {return symptoms.size();}


    public static class SymptomViewHolder extends RecyclerView.ViewHolder {
        TextView intensityText;
        TextView descriptionText;
        TextView dateText;
        TextView locationText;

        public SymptomViewHolder(@NonNull View itemView) {
            super(itemView);

            intensityText = itemView.findViewById(R.id.intensity_text);

            descriptionText = itemView.findViewById(R.id.symptom_description);

            dateText = itemView.findViewById(R.id.date_text);

            locationText = itemView.findViewById(R.id.location_text);
        }
    }
}

File: app/src/main/java/com/unnebulous/consultapronta/recyclerview/adapter/SymptomAdapter.kt
package com.unnebulous.consultapronta.recyclerview.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.unnebulous.consultapronta.database.Symptom
import com.unnebulous.consultapronta.databinding.CardSymptomBinding
import com.unnebulous.consultapronta.toBrazilianLocale

class SymptomAdapter: ListAdapter<Symptom, SymptomAdapter.SymptomViewHolder>(SymptomComparator()) {
	enum class SymptomViewType { COMPACT, DETAILED }

	lateinit var onClick: (Symptom) -> Unit

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SymptomViewHolder {
		val binding = CardSymptomBinding.inflate(
			LayoutInflater.from(parent.context),
			parent,
			false
		)

		return SymptomViewHolder(binding)
	}

	override fun onBindViewHolder(viewHolder: SymptomViewHolder, position: Int) {
		viewHolder.bind(getItem(position), onClick)
	}

	class SymptomComparator : DiffUtil.ItemCallback<Symptom>() {
		override fun areItemsTheSame(old: Symptom, new: Symptom) = old.id == new.id

		override fun areContentsTheSame(old: Symptom, new: Symptom) = old == new
	}

	class SymptomViewHolder(private val binding: CardSymptomBinding): RecyclerView.ViewHolder(binding.root) {
		fun bind(symptom: Symptom, onClick: (Symptom) -> Unit) {
			binding.apply {
				titleText.text = symptom.title
				descriptionText.text = symptom.description
				symptomDateText.text = symptom.dateTime!!.toBrazilianLocale()
				intensityView.setIntensity(symptom.intensity)
				placeView.setLocation(symptom.place)

				// Não existem ainda
				symptomAnnexesQuantityText.visibility = View.GONE
				editButton.visibility = View.GONE

				root.setOnClickListener { onClick(symptom) }
			}
		}
	}
}